### PR TITLE
Feature/jwt validation with jwks endpoint

### PR DESCRIPTION
**Pull request by haiko-ritense**
See issue https://github.com/valtimo-platform/valtimo-issues/issues/1

### CODE REVIEW

File: contract/src/main/java/com/ritense/valtimo/contract/security/jwt/provider/SecretKeyProvider.java [hunk 1]
```diff
@@ -24,6 +24,6 @@ public interface SecretKeyProvider {
 
     boolean supports(SignatureAlgorithm algorithm, Claims claims);
 
-    Key getKey(SignatureAlgorithm algorithm);
+    Key getKey(SignatureAlgorithm algorithm, String kId);
```
Comment (theo-ritense) [inline contract/src/main/java/com/ritense/valtimo/contract/security/jwt/provider/SecretKeyProvider.java:27]: Looks like a breaking change. Can you implement it in such a way that it's non-breaking?

Comment (haiko-ritense) [reply contract/src/main/java/com/ritense/valtimo/contract/security/jwt/provider/SecretKeyProvider.java:27]: no, I am afraid not. KeyID is necessary to get the correct key from the keyset.

File: keycloak-iam/src/main/java/com/valtimo/keycloak/autoconfigure/KeycloakAutoConfiguration.java [hunk 2]
```diff
@@ -53,6 +58,15 @@ public KeycloakSecretKeyProvider keycloakSecretKeyProvider(
         return new KeycloakSecretKeyProvider(secret);
     }
 
+    @Bean
+    @ConditionalOnMissingBean(JwksKeyCloakProvider.class)
+    @ConditionalOnProperty("valtimo.jwt.jwks")
+    public JwksKeyCloakProvider keycloakSecretKeyProviderForJwks(
+        @Value("${valtimo.jwt.jwks}") final String jwks
```
[the remark this answers is not in the document]
Comment (haiko-ritense) [reply keycloak-iam/src/main/java/com/valtimo/keycloak/autoconfigure/KeycloakAutoConfiguration.java:65]: If you got suggestions, I am open for them. JWKS is actually is know acronym. See info on https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-key-sets ->  The JSON Web Key Set (JWKS) is a set of keys containing the public keys used to verify any JSON Web Token (JWT) issued by the Authorization Server and signed using the RS256 [signing algorithm](https://auth0.com/docs/get-started/applications/signing-algorithms)

**Comment by haiko-ritense** [comment]
need some rework

**Pull request closed without merging**